client/allocrunner/taskrunner/getter/util_linux.go

```diff
@@ -0,0 +1,91 @@
+//go:build linux
+
+package getter
+
+import (
+	"fmt"
+	"path/filepath"
+	"syscall"
+
+	"github.com/hashicorp/nomad/helper/users"
+	"github.com/shoenig/go-landlock"
+)
+
+var (
+	// version of landlock available, 0 otherwise
+	version int
+
+	// userUID is the current user's uid
+	userUID uint32
+
+	// userGID is the current user's gid
+	userGID uint32
+)
+
+func init() {
+	v, err := landlock.Detect()
+	if err == nil {
+		version = v
+	}
+
+	userUID = uint32(syscall.Getuid())
+	userGID = uint32(syscall.Getgid())
+}
+
+// attributes returns the system process attributes to run
+// the sandbox process with
+func attributes() *syscall.SysProcAttr {
+	uid, gid := credentials()
+	return &syscall.SysProcAttr{
+		Credential: &syscall.Credential{
+			Uid: uid,
+			Gid: gid,
+		},
+	}
+}
+
+// credentials returns the UID and GID of the user the child process
+// will run as. On Linux systems this will be the nobody user if Nomad
+// is being run as the root user, or the user Nomad is being run as
+// otherwise.
+func credentials() (uint32, uint32) {
+	switch userUID {
+	case 0:
+		return users.NobodyIDs()
+	default:
+		return userUID, userGID
+	}
+}
+
+// minimalVars returns the minimal environment set for artifact
+// downloader sandbox
+func minimalVars(taskDir string) []string {
+	tmpDir := filepath.Join(taskDir, "tmp")
+	return []string{
+		"PATH=/usr/local/bin:/usr/bin:/bin",
+		fmt.Sprintf("TMPDIR=%s", tmpDir),
+	}
+}
+
+// lockdown isolates this process to only be able to write and
+// create files in the task's task directory.
+// dir - the task directory
+//
+// Only applies to Linux, when available.
+func lockdown(dir string) error {
+	// landlock not present in the kernel, do not sandbox
+	if !landlock.Available() {
+		return nil
+	}
```

Comment:
The go-landlock docs say a noop version is used when its not available, could we skip some detection logic and just always _try_ to use landlock? The API itself seems to be designed with that in mind.

If we do need to be careful to only call landlock when its available, maybe update the docs on the go-landlock side.

Not a big deal. I just want to make sure folks opportunistically using landlock in the future learn the right patterns.